source/module_hamilt_pw/hamilt_pwdft/kernels/cuda/stress_op.cu

```diff
@@ -353,6 +353,115 @@ __global__ void cal_vq_deri(
         tab, it, ib, tab_2, tab_3, table_interval, gnorm[idx]);
 }
 
+
+
+template <typename FPTYPE>
+__device__ void Simpson_Integral
+(
+    const int mesh,
+    FPTYPE * func,
+    const FPTYPE * rab,
+    FPTYPE &asum
+)
+{
+    asum = 0.00;
+	const size_t end = mesh-2;
+    for( size_t i=1; i!=end; i+=2 )
+    {
+		const double f1 = func[i]*rab[i];
+		asum += f1 + f1 + func[i+1]*rab[i+1];
+    }
+	const double f1 = func[mesh-2]*rab[mesh-2];
+	asum += f1+f1;
+	asum += asum;
+	asum += func[0]*rab[0] + func[mesh-1]*rab[mesh-1];
+	asum /= 3.0;
+    return;
+}// end subroutine simpson
+
+
+#define FOUR_PI 4.0 * 3.14159265358979323846
+template <typename FPTYPE>
+__global__ void cal_stress_drhoc_aux(
+        const FPTYPE* r, const FPTYPE* rhoc, 
+        const FPTYPE *gx_arr, const FPTYPE *rab, FPTYPE *drhocg, 
+        const int mesh, const int igl0, const int ngg, const double omega,
+        int type
+){
+    int idx = threadIdx.x + blockIdx.x * blockDim.x;
+
+    FPTYPE aux_d[1300];
```

Comment:
why 1300? Need a note in the code.

Reply:
This code had bugs in the initial commit, and the fix produced this 1300 array for testing. This will be removed in the next commit.